pkg/validate/sources.go

```diff
@@ -0,0 +1,49 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	build "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+)
+
+// SourcesRef implements RuntimeRef interface to add validations for `build.spec.sources` slice.
+type SourcesRef struct {
+	Build *build.Build // build instance for analysis
+}
+
+// ValidatePath executes the validation routine, inspecting the `build.spec.sources` path, which
+// contains a slice of BuildSource.
+func (s *SourcesRef) ValidatePath(ctx context.Context) error {
```

Comment:
Not sure if I miss something, but context seems to be unused.

```suggestion
func (s *SourcesRef) ValidatePath(_ context.Context) error {
```